src/commonMain/kotlin/com/ucasoft/kcron/builders/Builder.kt

```diff
@@ -82,29 +82,56 @@ class Builder(firstDayOfWeek: WeekDays = WeekDays.Monday) {
-                        for (minute in (partBuilders.getValue(CronPart.Minutes) as MinutesBuilder).minutes.filter { m -> m >= now.minute || hour > now.hour || day > now.dayOfMonth || month > now.monthNumber || year > now.year }) {
-                            for (seconds in (partBuilders.getValue(CronPart.Seconds) as SecondsBuilder).seconds.filter { s -> s > now.second || minute > now.minute || hour > now.hour || day > now.dayOfMonth || month > now.monthNumber || year > now.year }) {
-                                result.add(LocalDateTime(year, month, day, hour, minute, seconds))
-                                if (result.size == maxCount) {
-                                    return result
+    @DelicateIterableApi
+    fun asIterable(start: LocalDateTime): Iterable<LocalDateTime> {
+        return Iterable {
+            iterator {
+                for (year in (partBuilders.getValue(CronPart.Years) as YearsBuilder).years.filter { y -> y >= start.year }) {
+                    for (month in (partBuilders.getValue(CronPart.Months) as MonthsBuilder).months.filter { m -> m >= start.monthNumber || year > start.year}) {
+                        for (day in calculateDays(year, month, (partBuilders.getValue(CronPart.DaysOfWeek) as DaysOfWeekBuilder).daysOfWeek, (partBuilders.getValue(CronPart.Days) as DaysBuilder).days, start)) {
+                            for (hour in (partBuilders.getValue(CronPart.Hours) as HoursBuilder).hours.filter { h -> h >= start.hour || day > start.dayOfMonth || month > start.monthNumber || year > start.year }) {
+                                for (minute in (partBuilders.getValue(CronPart.Minutes) as MinutesBuilder).minutes.filter { m -> m >= start.minute || hour > start.hour || day > start.dayOfMonth || month > start.monthNumber || year > start.year }) {
+                                    for (seconds in (partBuilders.getValue(CronPart.Seconds) as SecondsBuilder).seconds.filter { s -> s > start.second || minute > start.minute || hour > start.hour || day > start.dayOfMonth || month > start.monthNumber || year > start.year }) {
+                                        yield(LocalDateTime(year, month, day, hour, minute, seconds))
+                                    }
                                 }
                             }
                         }
                     }
                 }
             }
         }
-
+    }
+    @Deprecated(
+        message = "Use asIterable(start: LocalDateTime) instead.",
+        replaceWith = ReplaceWith(
+            expression = "asIterable(start = TODO()).take(maxCount)",
+            imports = [
+                "kotlin.collections.take",
+            ]
+        )
+    )
+    @OptIn(DelicateIterableApi::class)
+    fun asIterable(maxCount: Int) : Iterable<LocalDateTime> {
```

Comment:
Why don't move the function code into `nextRunList` function directly instead?

Reply:
Just to make correct autoreplacement in your source code of your project w/o breaking the contract:

![image](https://github.com/Scogun/kcron-common/assets/9965210/f74c9fcc-2881-461a-96ef-4e3b3554fe6b)

Reply:
So, default value for `start` parameter will solve it, right?

Reply:
I think it would be better to left `asIterable(start: LocalDateTime)` w/o default parameter since it uses API backed by platform-specific implementation that:
- isn't required in general case (for example, when using to find runs from start of a day/month/year);
- could be invalid or out of sync with external trusted source (for example, due to invalid system timezone).

To achive the same effect an extension function could be used in your project:
```kotlin
fun Builder.asIterable(): Iterable<LocalDateTime> {
    val now = Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault())
    return asIterable(start = now)
}
```

P.S. I'd like to suggest not to use `kotlin.time.Clock` in the project and let developers take responsibility on how to provide current datetime :)

Reply:
But `LocalDateTime` is from **Kotlinx Datetime**. Not Java one! Otherwise, it wasn't possible to make the library as multiplatform one.
However, extract Datetime provider - it's **can be** a good idea. I will create a ticket for it.
But right now is more important to provide right backward compatibility.

Reply:
> But `LocalDateTime` is from **Kotlinx Datetime**. Not Java one! Otherwise, it wasn't possible to make the library as multiplatform one

Sure, I know about that.

> But right now is more important to provide right backward compatibility.

You're right, I'm just have some plans to contibute much more to the project and may overcomplicate changes in this pull request to support :)

Reply:
Changed in [f78076e2](https://github.com/Scogun/kcron-common/pull/20/commits/f78076e2c5962c2a6755893c97eb357afa04b524)